tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/server/AbstractIdeTest.java

```diff
@@ -149,6 +149,10 @@ static final public void resetPrintStreams() {
 	/** Deletes the test project in case it exists. */
 	@After
 	final public void deleteTestProject() {
+		// clear thread pools
+		languageServer.shutdown().join();
+		languageServer.getRequestManager().shutdown();
```

Comment:
Would a `joinServerRequests()` be an option here? (instead of lines 153, 154).
(except for the disadvantage that we would then always wait for those obsolete requests to complete instead of cancelling them)

Reply:
joinServerRequests would not shutdown the server. The shutdown should be faster than a joinServerRequests.